src/Analysis/Engine/Test/FluentAssertions/AnalysisValueAssertions.cs

```diff
@@ -95,6 +95,22 @@ public AndConstraint<TAssertions> HaveMembers(params string[] memberNames)
             return new AndConstraint<TAssertions>((TAssertions)this);
         }
 
+        public AndConstraint<TAssertions> NotHaveMembers(params string[] memberNames)
+            => NotHaveMembers(memberNames, string.Empty);
+
+        public AndConstraint<TAssertions> NotHaveMembers(IEnumerable<string> memberNames, string because = "", params object[] reasonArgs) {
+            var actualNames = Subject.GetAllMembers(((ModuleScope)OwnerScope.GlobalScope).Module.InterpreterContext).Keys.ToArray();
```

Comment:
`((ModuleScope)OwnerScope.GlobalScope).Module.InterpreterContext` is used in several places. Let's extract it into helper method.